File: Tutorial2-2Mockito-BDD/src/main/java/com/smarttoolfactory/tutorial2_2mockito_bdd/model_login_manager/LoginManagerImpl.java
package com.smarttoolfactory.tutorial2_2mockito_bdd.model_login_manager;

public class LoginManagerImpl {

    ILoginManager loginManager;

    public User login(String email, String password) {
        return loginManager.login(email, password);
    }

    public String validate(String email, String password) {
        return loginManager.validate(email, password);
    }
}

File: Tutorial4-1Kotlin/src/main/java/com/smarttoolfactory/tutorial4_1kotlin/model_login_manager/LoginManagerImpl.kt
package com.smarttoolfactory.tutorial4_1kotlin.model_login_manager

class LoginManagerImpl {
    var loginManager: ILoginManager? =
        null

    fun login(
        email: String?,
        password: String?
    ): User? {
        return loginManager!!.login(email, password)
    }

    fun validate(email: String?, password: String?): String? {
        return loginManager!!.validate(email, password)
    }
}